GitUI/UserControls/WarningHeader.cs

```diff
@@ -0,0 +1,193 @@
+﻿using System;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using GitCommands;
+using GitUI.CommandsDialogs;
+using Microsoft.VisualStudio.Threading;
+
+namespace GitUI.UserControls
+{
+    public partial class WarningHeader : UserControl
```

Comment:
If a maintainer browses the codebase and sees a file named `WarningHeader` the maintainer is unlikely to understand what this type is.
I suggest to rename to something like `RebaseProgressWarning` as such name communicates the control's purpose clearer.

Reply:
Sure, though it can display more than just rebase progress. I wasn't sure of the right word to capture all the states we display, ActionProgressHeader? RepositoryStateHeader?

Reply:
RevisionGridWarningHeader? Or RevisionGridProgressHeader ?

Reply:
I wouldn't call it "header" at all. Today it is placed above the revision grid, tomorrow we may move it to another place. I wouldn't want to rename the control.

Reply:
`InteractiveGitOperationProgressNotification`?

Reply:
I tried to keep it a bit shorter so went with InteractiveGitActionControl